async/logic/Async4BackendRest_DestinationAPI_TopicCheckLogic.java

```diff
@@ -0,0 +1,543 @@
+			
+			destinationHttpMethod = jsonObject.getString("destinationhttpmethod");
+			
+			if(receivemessage.contains("destinationrequestheader")) {
+				destinatioRrequestHeader = jsonObject.getString("destinationrequestheader");
+				
+				String[] splitresult = destinatioRrequestHeader.split(":", 0);
+				destinatioRrequestHeaderKey = splitresult[0];
+				
+				destinatioRrequestHeaderValue = splitresult[1];
+				
+			}
+			
+			if(receivemessage.contains("destinationrequestbody")) {
+				// URLDecodeする
+				destinationRequestBody = URLDecoder.decode(jsonObject.getString("destinationrequestbody"), "utf-8");
+				
+			}
+			
+			executionTimes.append(jsonObject.getString("executiontimes"));
+			
+			URL url = new URL(destinationUri);
+			
+			HttpURLConnection urlConnection = null;
+			
+			urlConnection = (HttpURLConnection) url.openConnection();
+			
+			urlConnection.setRequestMethod(destinationHttpMethod);
+			
+			urlConnection.setUseCaches(false);
+			
+			urlConnection.setDoOutput(true);
+			
+			if(receivemessage.contains("destinationrequestheader")) {
+				urlConnection.setRequestProperty(destinatioRrequestHeaderKey, destinatioRrequestHeaderValue);
+			}
+			
+			OutputStreamWriter out = new OutputStreamWriter(
+					new BufferedOutputStream(urlConnection.getOutputStream()));
+			
+			if(receivemessage.contains("destinationrequestbody")) {
+				out.write(destinationRequestBody.toString());
+			}
+			
+			out.close();
+			
+			destinationStatus.append(urlConnection.getResponseCode());
+			
+			BufferedReader br = null;
+			
+			try {
+				// レスポンスを受け取る
+				br = new BufferedReader(new InputStreamReader(urlConnection.getInputStream(), "UTF-8"));
+				
+				String line;
+				
+				while((line = br.readLine()) != null) {
+					destinationResponseBody.append(line);
+				}
+				
+				br.close();
+				
+			// 200番台以外の場合
+			} catch(IOException e) {
+				
+				BufferedReader errbr = null;
+				try {
+					
+					errbr = new BufferedReader(new InputStreamReader(urlConnection.getErrorStream(), "UTF-8"));
+
+					String errline;
+					
+					while((errline = errbr.readLine()) != null) {
+						destinationResponseBody.append(errline);
+					}
+					
+					return GeneralErrorRes(ErrorResponseEnum.E500);
+				} catch(Exception e1) {
+					
+					e1.printStackTrace();
+				} finally {
+					if(errbr != null) {
+						try {
+							errbr.close();
+						} catch (IOException e2) {
+							
+							e2.printStackTrace();
+						}
+					}
+				}
+			} catch(Exception e) {
+				
+				e.printStackTrace();
+			} finally {
+		    	if (br != null) {
+		    		br.close();
+		    	}
+		    }
+		} catch(Exception e) {
+			
+        	e.printStackTrace();
+			
+	    	return GeneralErrorRes(ErrorResponseEnum.E500);
+	    } 
+        	    
+	    return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+	}
+
+	private Response CreateRetryMessage(String receivemessage, StringBuilder retrymessage,  StringBuilder executiontimes) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 76 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.